x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java

```diff
@@ -57,6 +58,7 @@ public NativeAnalyticsProcessFactory(Environment env,
                                          DataFrameAnalyticsAuditor auditor) {
         this.env = Objects.requireNonNull(env);
         this.nativeController = Objects.requireNonNull(nativeController);
+        this.clusterService = Objects.requireNonNull(clusterService);
```

Comment:
Seems to me that we don't need to keep a reference to the clusterService. we could have `this.nodeName = Objects.requireNonNull(clusterService).getNodeName()` as the internal `nodeName` is a `final` in the `ClusterService` object.